### PR TITLE
[UI|Console] added console option to specify move completed folder

### PR DESCRIPTION
**Pull request by hukal**
This option helps to split items to different folders. Useful for Plex media server for example.
Would be nice to see in ubuntu ppa.

### CODE REVIEW

**Comment by DjLegolas** [comment]
Can you fix the failed test (regarding coding style)?
In addition, please try to add a test for this.

**Comment by DjLegolas** [comment]
I'll start with the failed test: The current coding style is new and is being validated with a dedicated test.
It is something @cas-- added some time ago.
The best way to check this, is to install `tox` and run `tox -e lint`. It will run the same test and, IIRC, modify the file as needed (note: don't run only once, but after each fix).

About adding new test, you are correct.. There are no tests for the console ui... So just start one from scratch, and test only what you added for now.
You can ask here for help.

And look at the [running tests](https://deluge.readthedocs.io/en/develop/contributing/testing.html) docs.

Finally, you closed this PR 😆

**Comment by DjLegolas** [comment]
A little update, there are tests for the Console.
Look at the `test_ui_entry.py` file. In it there are most of the UI entry tests, including `ConsoleUIBaseTestCase` class, which tests some the Console's functionality.

File: deluge/tests/test_ui_console_add.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+from __future__ import unicode_literals
+
+import argparse
+
+from deluge.tests.test_ui_entry import ConsoleUIBaseTestCase
+from deluge.ui.console.cmdline.commands.add import Command
+
+
+class UICommonTestCase(ConsoleUIBaseTestCase):
```
Comment (DjLegolas) [inline deluge/tests/test_ui_console_add.py]: should inherit from `basetest.BaseTestCase` as well for the tests to be run.

**Comment by hukal** [comment]
Hi @DjLegolas ,
I've fixed inheritance and naming.
But it looks like I can't find a proper way to make an end to end test.

Things I have tried:
1. Directly call `handle` method of `Add` command.
In this case I'm getting _"ConsoleUI component is not registered"_ error.
Event if I register it (mock?) I can't do anything since `Add` command requires daemon.

2. In `test_ui_entry.py` we have `test_console_command_status` which works with daemon.
I was able to make a similar one to test 'add' command but this command returns only "Torrent added" without any id/hash. Due to this fact I can't validate the file in the way it validated in `test_add_torrent_file` from `test_core`.

3. In addition to  point 2 I've tried to run `status` command right after `add` in one method. (I was hoping to grep hash from there). It also caused errors. (Am I correct that current test structure supports 1 console call per method?)

Could you please advise how to test it?

Thanks

File: deluge/tests/test_ui_console_add.py (hunk 1 above)
Comment (hukal) [reply deluge/tests/test_ui_console_add.py]: Done.

**Comment by DjLegolas** [comment]
Ok, I managed to run the test by adding a call to `status` command between `add` and `manage` commands. In addition, I added a test for the `add` command itself.
Changes pushed it to @hukal fork.
I'm sure it's not the best way to handle this, but for now it will do. I hope 🤣

**Comment by hukal** [comment]
Thanks @DjLegolas !
I've checked the tests locally - all green.
I'm only getting a warning: 
`common.py:270: PendingDeprecationWarning: dist() and linux_distribution() functions are deprecated in Python 3.5`
My local env is on Python 3.6.7

**Comment by hukal** [comment]
@DjLegolas Build error, again.
`INTERNALERROR> PluginValidationError: unknown hook 'pytest_namespace' in plugin <module 'pytest_twisted' from 'c:\projects\deluge\.tox\py27\lib\site-packages\pytest_twisted.py'>`
Is it because of my changes?

**Comment by hukal** [comment]
OK, let's wait. Thanks!

**Comment by DjLegolas** [comment]
@cas-- What do you say? Want to merge this?

**Comment by cas--** [comment]
It looks good, I will get around to reviewing and merging :+1:

**Comment by cas--** [comment]
Thanks for this and the test :+1: 

I have made some changes (-m instead of -c) and merged it: f885edd7fc85f

**Pull request closed without merging**